File: MATLAB/functions/IS_PAPER_PARAMETERS.m
function matches = IS_PAPER_PARAMETERS(params)
%IS_PAPER_PARAMETERS Check the structural parameterization associated with the paper's T.
reference = struct('alpha_R',.66,'alpha_C',.85,'beta_R',0,'beta_dens_R',-.10535545480050244,'beta_C',.03,'tau_R',.016,'tau_C',.014,'x_core_R',1,'x_core_C',1,'omega_R',.07,'omega_C',.03,'theta_R',.55,'theta_C',.5,'c_R',150,'c_C',150,'a_bar_R',2,'a_bar_C',2,'r_a',50,'ell',.5,'zeta',6.4,'T',2.9605069278556773); % Exclude city moments, inferred rural utility and policy limits.
fields = fieldnames(reference); matches = true; % Accept only the paper structural values, allowing floating-point roundoff.
for k=1:numel(fields)
    name=fields{k}; matches=matches && isfield(params,name) && isscalar(params.(name)) && abs(params.(name)-reference.(name))<=1e-10*max(1,abs(reference.(name))); % Check every structural parameter.
end
end

File: Python/functions/is_paper_parameters.py
"""Identify the structural parameterization associated with the paper's T."""

import numpy as np


PAPER_PARAMETERS = {
    "alpha_R": 0.66, "alpha_C": 0.85, "beta_R": 0.0,
    "beta_dens_R": -0.10535545480050244, "beta_C": 0.03,
    "tau_R": 0.016, "tau_C": 0.014, "x_core_R": 1.0, "x_core_C": 1.0,
    "omega_R": 0.07, "omega_C": 0.03, "theta_C": 0.5, "theta_R": 0.55,
    "c_R": 150.0, "c_C": 150.0, "a_bar_R": 2.0, "a_bar_C": 2.0,
    "r_a": 50.0, "ell": 0.5, "zeta": 6.4, "T": 2.9605069278556773,
}


def is_paper_parameters(params):
    """Return True when every paper structural parameter agrees to numerical tolerance."""
    return all(name in params and np.isclose(params[name], value, rtol=1e-10, atol=1e-10)
               for name, value in PAPER_PARAMETERS.items())